### PR TITLE
[refactor] measureCat. [feature] derived variable. 

### PR DESCRIPTION
**Pull request by DougManuel**
Two items that address #14, #32, #38

1) Revert/refactor wwPara from Measurement.csv to Sample.csv. wwPara are measures that are taken at the time the sample is taken and represent features of that sample such as temperate, flow rate. Measurement.csv represents one or more viral and reference measurements taken on a sample. These represent lab assays on a sample, rather than the site. 

2) Proposed description of derived measure.

This PR is a partial description with additional parameters and derived measure that can be added if the description is sound.

### CODE REVIEW

**Comment by hswerdfe** [comment]
There is no place to put the value for the wwPara (only the type) that I see. and with it in the sample table, it can only be 1 to 1 with the sample. 

I think I would propose one of two thing, leave wwParam in the Measurement table, or build out the sample table to include columns for every wwParam we think is important.
 - Temp
 - PH
 - bod
 - cod
 - flow 
 - tss

**Comment by DougManuel** [comment]
Yes. You are correct. I'll move the wwPara back to measurement. It isn't a perfect solution but the best option.

**Comment by hswerdfe** [comment]
I just thought of a third option, is build a new table.
this table would be for "simple" measures, like things where the lab and assay are not as important, and don't need to be recorded.

**Comment by DougManuel** [comment]
Item 1 has been resolved in  PR https://github.com/Big-Life-Lab/covid-19-wastewater/pull/44.

**Comment by DougManuel** [comment]
Item 2. Summary of discussion with @hswerdfe. 

We agreed that we should have derived or summary measures specified in the metadata. This is a somewhat unusual approach given that most database structures do not consider derived or summary measures as part of a database structure or dictionary. 

However, lab instruments do output these summary measures and they are the main measures that are currently reported by most labs worldwide and will be used in analyses or publicly reported. Furthermore, there are statistical language metadata systems such as [SDMX](https://linked-statistics.github.io/SDMX-Metadata/sdmx-metadata.html).

On reflection from our discussion, regardless of the approach, we should:

1)  have a list of unambiguous derived or summary measures that are commonly used for the wide table format. This list will reduce errors for people creating a wide list and provide naming recipes.

2) have an approach to name new derived or summary measures. Sharing data derived measures is common and therefore should be supported in the metadata or dictionary to improve reproducible research and reports. 

We discussed three options. 

1) The first option is what is identified in this PR. 

Advantages: 
- works for a range of use cases.
- it is explicit. 
- the variable names/headers could be machine-generated and machine-actionable. 

Disadvantages:
- fairly complex. New derived measures would often be created by a user and there could be errors in variable names.
- not unambitious. `N1-N2-mean` is the same as `N2-N1-mean`. However, there is an approach to address this issue in PR https://github.com/Big-Life-Lab/covid-19-wastewater/pull/44

2) The second option is to keep the approach largely unchanged. 

Advantage:
- practical - what is currently being done by labs.

Disadvantage 
- potential proliferation of categories in `measureCat` and `measureUnit`. 
- `measureUnit` has two concepts. Units for single measures and standardization or normalization units.

3) The Third option is the creation of a standardization or normalization approach. Take out standardization from `measureUnit`.
On reflection, I don't think this approach is an improvement over option 2. We will still need `measureUnit` for all standardization metrics.

**Comment by hswerdfe** [comment]
I think it all depends on what can be agreed on at the lab level, and at the funding level. 
there are a couple layers of normalization and aggregation that go on in the labs
 * average multiple runs through the PCR machine for the same target
 * normalize the target to a standard curve
 * normalize the results to account for sample extracted
 * normalize the results to account for measurement loss (MHV)
 * normalize the results to account for PPMV or CrA
 * moving averages in time to smooth out variations

depending which of these values is accounted for and when and how and in which order you may get very different results to report.

Fundamentally does this project want to account for all of these steps or some of them, or none of them ?

**Comment by hswerdfe** [comment]
Ok thinking about it and where we are now I propose this is what we care about

 * average multiple runs through the PCR machine for the same target  <- YES
 * normalize the target to a standard curve  <- NO
 * normalize the results to account for sample extracted <- NO
 * normalize the results to account for measurement loss (MHV)  <- NO
 * normalize the results to account for PPMV or CrA <- YES
 * moving averages in time to smooth out variations <- YES

basically that would mean for any given sample we would take things like 

N1_1, N1_2, PPMV_1, PPMV_2
but they would all be in actual units **copied per ml**, none of this cycles stuff, then we could do the average across multiple runs through the PCR for each of them and the normalization against PMMV or CrA, in calculations.

this basically means we only accept targets and we wouldn't accept N1N2Avg anymore.


thoughts
@DougManuel

**Comment by DougManuel** [comment]
Virtually all the information that is transmitted around the world is average data. None raw data. I think we need to be pragmatic and offer an approach for reporting summary or derived measures like averages and SDs. It will be a mess if we don't. 

Labs and regions will transmit a lot of data with their own formats and we'll be back to low coverage and/or incompatible data.

Transmitting only raw data and not averages is the ideal approach, but that is up to labs and programs -- not a metadata standard.

**Comment by DougManuel** [comment]
Ideally, we would have an approach to describe different forms of normalized data. However, I think we need to prioritize with normalized or standardization that is commonly transmitted from labs into a common data platform. 

We certainly would like to have an approach that can be expanded to various forms of normalized data. The approach that we decide now should be robust for future additions.

**Comment by hswerdfe** [comment]
@DougManuel  getting back to the issue of waste water treatment plant parameters. I am now starting to think that its a good idea   to separate them from the covid data. they are simply not on the same scale of complexity.

I propose we move them to the sample table 1 column each.
its the easiest.

**Comment by DougManuel** [comment]
@hswerdfe Can you review the PR again. 

Update:
1. Revert/refactor wwPara from Measurement.csv. These changes have been removed and differed the folks working on WWTP parameters.

2. Proposed description of derived measure. This PR now only proposes a description of derived measures.

**Review by hswerdfe — COMMENTED** [review]
It looks like the **measurement** tables is using the new short naming convention.

The **sample** tables is not using the short naming convention.

**Site** table does use short names 


**AssayMethod** table has `Assay.ID` but should jus have `ID`

**Review by hswerdfe — COMMENTED** [review]
Derived summary transform section looks fine

**Review by hswerdfe — APPROVED** [review]
